### PR TITLE
Content security policy

### PR DESCRIPTION
**Pull request by mattonfoot**
### Context
The previous version of the Content Security Policy blocked the API docs and didn't report correctly. 

### Changes proposed in this pull request
This change adds a very permissive CSP policy backed by a restrictive "report only" policy. It also addresses the lack of screen shots on the API Docs that would have identified the blocked content.

### Testing
- Cypress tests for sandbox modified
- csp_report_controller_spec checks logging for expected logs

### Review Checks
- [ ] API docs page have visual tests via cypress + percy
- [ ] testing is of sufficient level - Q: do we need tests that will break if the CSP headers change?
- [ ] Check the CSP to see if anything specific stands out as missing

### How can I view this in a review app?
The developer console will need to be up. As you browse the website you should see no CSP report errors displayed or the only ones you find should be labeled "report only".

### CODE REVIEW

File: spec/cypress/integration/Sandbox.feature [hunk 1]
```diff
@@ -13,5 +13,4 @@ Feature: Sandbox landing page
     Given I am on "the sandbox landing page" page
     When I click on "link" containing "Review our API documentation"
     Then I should be on "API Documentation" page
-    And the page should be accessible
```
Comment (timabell) [inline spec/cypress/integration/Sandbox.feature]: I think the front page should still be checked for a11y

File: config/initializers/secure_headers.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  config.x_download_options = "noopen"
+  config.x_permitted_cross_domain_policies = "none"
+  config.referrer_policy = %w[origin-when-cross-origin strict-origin-when-cross-origin]
+
+  google_analytics = %w[www.google-analytics.com ssl.google-analytics.com *.googletagmanager.com tagmanager.google.com *.googleusercontent.com *.gstatic.com]
+
+  config.csp = {
+    default_src: %w[* data: 'unsafe-inline'],
+    script_src: %w[* 'unsafe-inline'],
+  }
```
Comment (ethanmills) [inline config/initializers/secure_headers.rb:17]:
```suggestion
  config.csp = SecureHeaders::OPT_OUT
```

**Review by ethanmills — CHANGES_REQUESTED** [review]
Let's be extra safe about this one

File: config/initializers/secure_headers.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# rubocop:disable Lint/PercentStringArray
+SecureHeaders::Configuration.default do |config|
+  config.x_frame_options = "DENY"
```
Comment (ethanmills) [inline config/initializers/secure_headers.rb:5]: Is this going to affect GA? I think that falls back to a frame for no-js

Comment (mattonfoot) [reply config/initializers/secure_headers.rb:5]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options it stops your website rendering in an iframe

Comment (ethanmills) [reply config/initializers/secure_headers.rb:5]: We have

```
    <!-- Google Tag Manager (noscript) -->
    <noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-N58Z5PG"
    height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
    <!-- End Google Tag Manager (noscript) -->
```

**Review by ethanmills — APPROVED** [review]
I've check the review app, I'm seeing a report-only (only) policy, and violations are successfully making their way through to Logit. Slight concern about GA with frames, but it should be a minor of users

File: config/initializers/secure_headers.rb (hunk 3 above)
Comment (mattonfoot) [reply config/initializers/secure_headers.rb:5]: thats us including an iframe not our website in an iframe on another website

Comment (mattonfoot) [reply config/initializers/secure_headers.rb:5]: So I think it is these lines that will cause issues for the google iframe 

frame_ancestors: %w['self'],
frame_src: %w['self'],

So will do some investigation

Comment (ethanmills) [reply config/initializers/secure_headers.rb:5]: Ah gotcha. Looks good then

Comment (ethanmills) [reply config/initializers/secure_headers.rb:5]: Those are both report-only, so we can mop that up with the rest

File: config/initializers/secure_headers.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  config.x_content_type_options = "nosniff"
+  config.x_xss_protection = "1; mode=block"
```
Comment (ethanmills) [inline config/initializers/secure_headers.rb:7]: Conflicts with #604

Comment (mattonfoot) [reply config/initializers/secure_headers.rb]: Good spot, adjusted to suit PEN test

**Pull request merged**